data/src/client.rs

```diff
@@ -505,42 +505,44 @@ impl Client {
                             && !message::is_action(text)
                         {
                             if let Some(query) = ctcp::parse_query(text) {
-                                match query.command {
-                                    "CLIENTINFO" => {
-                                        let _ = self.handle.try_send(command!(
+                                if matches!(&message.command, Command::PRIVMSG(_, _)) {
+                                    match query.command.as_ref() {
+                                        "CLIENTINFO" => {
+                                            let _ = self.handle.try_send(command!(
                                             "NOTICE",
-                                            user,
+                                            user.nickname().to_string(),
                                             "\u{1}CLIENTINFO ACTION CLIENTINFO DCC PING SOURCE VERSION\u{1}"
                                         ));
+                                        }
+                                        "PING" => {
+                                            let _ = self.handle.try_send(command!(
+                                                "NOTICE",
+                                                user.nickname().to_string(),
+                                                format!("\u{1}PING {}\u{1}", query.params)
+                                            ));
```

Comment:
I smell a `ctcp::format` method which takes a `ctcp::Command`, `User` and `&str` / params and formats these IRC messages so we don't have to define it here and accidentally miss `\u{1}` or like before, missing the `COMMAND` in the response